admin/build-doc

```diff
@@ -6,45 +6,52 @@ TOPDIR=`pwd`
 
 install -d -m0755 build-doc
 
-if command -v dpkg >/dev/null; then
-    packages=`cat ${TOPDIR}/doc_deps.deb.txt`
-    for package in $packages; do
-    if [ "$(dpkg --status -- $package 2>&1 | sed -n 's/^Status: //p')" != "install ok installed" ]; then
-        # add a space after old values
-        missing="${missing:+$missing }$package"
-    fi
-    done
-    if [ -n "$missing" ]; then
-        echo "$0: missing required packages, please install them:" 1>&2
-        echo "sudo apt-get install -o APT::Install-Recommends=true $missing" 1>&2
-        exit 1
-    fi
-elif command -v yum >/dev/null; then
-    for package in python-devel python-pip python-virtualenv doxygen ditaa ant libxml2-devel libxslt-devel Cython graphviz; do
-	if ! rpm -q --whatprovides $package >/dev/null ; then
-		missing="${missing:+$missing }$package"
-	fi
-    done
-    if [ -n "$missing" ]; then
-        echo "$0: missing required packages, please install them:" 1>&2
-        echo "yum install $missing"
-        exit 1
-    fi
-else
-    for command in virtualenv doxygen ant ditaa cython; do
-	command -v "$command" > /dev/null;
-	ret_code=$?
-	if [ $ret_code -ne 0 ]; then
-            # add a space after old values
-	    missing="${missing:+$missing }$command"
-	fi
-    done
-    if [ -n "$missing" ]; then
-	echo "$0: missing required command, please install them:" 1>&2
-	echo "$missing"
-	exit 1
-    fi
-fi
+source /etc/os-release
+case $ID in
+    debian|ubuntu|devuan)
```

Comment:
i am seeing 
```
+ source /etc/os-release
./do_cmake.sh: 9: ./do_cmake.sh: source: not found
```

in our xenial building host. for instance, see https://jenkins.ceph.com/job/ceph-pull-requests/11113/consoleFull .